stages/org.osbuild.cloud-init

```diff
@@ -142,12 +142,35 @@ SCHEMA = r"""
 """
 
 
+# Class representing the 'datasource_list' configuration option,
+# allowing to define a custom YAML dumper for it.
+class DatasourceList(list):
+    pass
+
+
+# Dedicated YAML dumper for the `DatasourceList` class.
+# Make sure that the `DatasourceList` always uses flow style.
+# https://cloudinit.readthedocs.io/en/latest/reference/base_config_reference.html#datasource-list
+# Specifically, it says for the `datasource_list` key:
+# "This key is unique in that it uses a subset of YAML syntax.
+#  It requires that the key and its contents, a list,
+#  must share a single line - no newlines."
+def datasourcelist_presenter(dumper, data):
+    return dumper.represent_sequence("tag:yaml.org,2002:seq", data, flow_style=True)
+
+
 # Writes the passed `config` object as is into the configuration file in YAML format.
 # The validity of the `config` content is assured by the SCHEMA.
 def main(tree, options):
     filename = options.get("filename")
     config = options.get("config", {})
 
+    datasource_list = config.get("datasource_list")
+    if datasource_list:
+        config["datasource_list"] = DatasourceList(datasource_list)
+
```

Comment:
(super super nitpick) I would remove the \n here and then all of this block deals with the datasource_list formating (but of course that is just personal taste).

Reply:
I was actually thinking about the same thing, but then went with adding the custom representer unconditionally. But technically, adding it conditionally should not be an issue. If you prefer it, I have nothing against that...

Reply:
I really just meant the one newline here (the \n) to have all the lines that work on datasource_list in the same (horizontal) block. Just:
```python
diff --git a/stages/org.osbuild.cloud-init b/stages/org.osbuild.cloud-init
index 63ec2cf0..aa986948 100755
--- a/stages/org.osbuild.cloud-init
+++ b/stages/org.osbuild.cloud-init
@@ -168,7 +168,6 @@ def main(tree, options):
     datasource_list = config.get("datasource_list")
     if datasource_list:
         config["datasource_list"] = DatasourceList(datasource_list)
-
     yaml.add_representer(DatasourceList, datasourcelist_presenter)
 
     config_files_dir = f"{tree}/etc/cloud/cloud.cfg.d"
```
or am I maybe misread your comment (or the code)?

Reply:
Oh, OK. I misunderstood your comment. "if statement block" vs. the "horizontal block" as you just pointed out. LGTM.

Reply:
done